components/Habit.tsx

```diff
@@ -0,0 +1,123 @@
+import { css } from '@emotion/react';
+import Image from 'next/image';
+import { Entry, Habit as THabit } from '../src/types/habits';
+
+const styles = {
+  container: css({
+    maxWidth: '500px',
+    margin: '0 auto'
+  }),
+
+  content: css({
+    display: 'flex',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    backgroundColor: '#333',
+    margin: '12px 0px',
+    padding: '12px',
+    gap: '8px'
+  }),
+
+  middleContainer: css({
+    display: 'flex',
+    flexDirection: 'column',
+    textAlign: 'left',
+    minWidth: '30%'
+  }),
+
+  deadline: css({
+    color: '#91E220'
+  }),
+
+  streakContainer: css({
+    display: 'flex',
+    flexDirection: 'row',
+    alignItems: 'center',
+    maxWidth: '100px',
+    padding: '0px 12px'
+  }),
+
+  streakText: css({
+    color: '#fad15c',
+    paddingLeft: '4px'
+  })
+};
+
+// Get the entry for today from the entries history
+const getEntriesForDay = (entries: Entry[], date: Date) =>
+  entries.filter(
+    ({ completionDate }) =>
+      completionDate.toDateString() === date.toDateString()
+  );
+
+const getEntriesForToday = (entries: Entry[]) =>
+  getEntriesForDay(entries, new Date());
+
+/**
+ * Check how much of the target habit has been completed
+ */
+const checkCompletionQuantity = (entries: Entry[]) =>
+  entries.reduce((total, entry) => (total += entry.quantity), 0);
```

Comment:
👌 

you don't actually want/need `+=` here, since it unwraps to
`total = total + entry.quantity`. The reason you don't want this is because the return value (`total + entry.quantity`) will be the next `total` value

Reply:
A mistake from leaving the code as it was when changing the forEach() loop to a reducer. Fixed.